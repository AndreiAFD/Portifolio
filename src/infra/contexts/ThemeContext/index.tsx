import React, { createContext, useState } from "react";
import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../core/theme/theme";

enum ThemeEnum {
  light = "light",
  dark = "dark",
}

interface AppThemeContextProps {
  theme?: string;
  handleChangeTheme?: (theme: ThemeEnum) => void;
}

export const AppThemeContext = createContext<AppThemeContextProps>({});

export const AppThemeContextProvider = ({ children }: { children: any }) => {
  const [actualTheme, setActualTheme] = useState<ThemeEnum>(ThemeEnum.light);

  const chakraTheme = extendTheme({
    colors: {
      ...theme[actualTheme],
    },
  });

  const handleChangeTheme = (nextTheme: ThemeEnum) => {
    setActualTheme(nextTheme);
  };

  return (
    <AppThemeContext.Provider
      value={{
        theme: actualTheme,
        handleChangeTheme,
      }}
    >
      <ThemeProvider theme={theme[actualTheme] as typeof theme["dark"]}>
        <ChakraProvider theme={chakraTheme}>{children}</ChakraProvider>
      </ThemeProvider>
    </AppThemeContext.Provider>
  );
};
