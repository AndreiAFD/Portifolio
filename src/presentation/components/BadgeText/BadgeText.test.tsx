import React from "react";
import { render, screen } from "@testing-library/react";
import { BadgeText } from ".";
import { AppThemeContextProvider } from "../../../infra/contexts/ThemeContext";

const ComponentToTest = () => {
  return (
    <AppThemeContextProvider>
      <BadgeText color="blue">Testing</BadgeText>
    </AppThemeContextProvider>
  );
};

describe("BadgeText component", () => {
  it("should render correctly", () => {
    render(<ComponentToTest />);
    const badgeRendered = screen.getByTestId("badge-text-testid");
    expect(badgeRendered).toMatchSnapshot();
  });
});
