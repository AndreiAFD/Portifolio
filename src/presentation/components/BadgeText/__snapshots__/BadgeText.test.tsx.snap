// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeText component should render correctly 1`] = `
<p
  class="chakra-text css-63qbpm"
  data-testid="badge-text-testid"
>
  Testing
</p>
`;
